src/infra/core/src/ingest/fetch_service/core.rs

```diff
@@ -46,6 +46,48 @@ pub struct FetchService {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+pub(crate) fn template_string<'a>(
+    s: &'a str,
+    lookup_fn: &dyn Fn(&str) -> Result<String, PollingIngestError>,
+) -> Result<Cow<'a, str>, PollingIngestError> {
+    let mut s = Cow::from(s);
+    let re_tpl = regex::Regex::new(r"\$\{\{([^}]*)\}\}").unwrap();
+    let re_env =
+        regex::Regex::new(r"^env\.([a-zA-Z-_0-9]+)\s?(?:\|\|\s+([a-zA-Z-_0-9]+))?$").unwrap();
+
+    loop {
+        if let Some(ctpl) = re_tpl.captures(&s) {
+            let tpl_range = ctpl.get(0).unwrap().range();
```

Comment:
Unless in tests, prefer `.expect("msg")` style to `.unwrap()`. It will be easier to locate the error by a more specific text.